frappe/core/doctype/system_settings/system_settings.js

```diff
@@ -27,3 +42,89 @@ frappe.ui.form.on("System Settings", "enable_two_factor_auth", function(frm) {
 		frm.set_value("bypass_restrict_ip_check_if_2fa_enabled", 0);
 	}
 });
+
+frappe.ui.form.on("System Settings", "enable_backup_encryption", function (frm) {
+    if ((frm.doc.enable_backup_encryption && frm.doc.new_password_encryption != frm.doc.confirm_password_encryption) || !frm.doc.confirm_password_encryption) {
+
+        frm.fields_dict.new_password_encryption.df.reqd = 1;
+        frm.fields_dict.new_password_encryption.df.hidden = 0;
+        frm.refresh_field("new_password_encryption");
+        frm.fields_dict.confirm_password_encryption.df.reqd = 1;
+        frm.fields_dict.confirm_password_encryption.df.hidden = 0;
+        frm.refresh_field("confirm_password_encryption");
+
+    } else {
+        frm.doc.new_password_encryption = "";
+        frm.doc.confirm_password_encryption = "";
+        frm.doc.password_encryption = "";
+        frm.fields_dict.new_password_encryption.df.reqd = 0;
+        frm.refresh_field("new_password_encryption");
+        frm.fields_dict.confirm_password_encryption.df.reqd = 0;
+        frm.refresh_field("confirm_password_encryption");
+        cur_frm.fields_dict.confirm_password_encryption.$input[0].onchange = null;
+
+    }
+    cur_frm.fields_dict.confirm_password_encryption.$input[0].onchange = function () {
+
+        if (frm.doc.new_password_encryption == frm.doc.confirm_password_encryption) {
+            $('input[data-fieldname="confirm_password_encryption"]')[0].style = "border-color: #08c708"
+
+        } else {
+            $('input[data-fieldname="confirm_password_encryption"]')[0].style = "border-color: red"
+
+        }
+
+    }
+
+});
+
+frappe.ui.form.on("System Settings", "change_encryption_password", function (frm) {
+    if (frm.doc.enable_backup_encryption && frm.doc.change_encryption_password) {
+        frm.fields_dict.new_password_encryption.df.reqd = 1;
+        frm.fields_dict.new_password_encryption.df.hidden = 0;
+        frm.refresh_field("new_password_encryption");
+        frm.fields_dict.confirm_password_encryption.df.reqd = 1;
+        frm.fields_dict.confirm_password_encryption.df.hidden = 0;
+        frm.refresh_field("confirm_password_encryption");
+        cur_frm.fields_dict.new_password_encryption.input_area.children[0].value = "******";
+        cur_frm.fields_dict.confirm_password_encryption.input_area.children[0].value = "******";
+
+
+    } else {
+        frm.fields_dict.new_password_encryption.df.reqd = 0;
+        frm.fields_dict.new_password_encryption.df.hidden = 1;
```

Comment:
Please look at [this line](https://github.com/frappe/frappe/pull/7880/files#diff-5ccf65b34d45e64504b1a1a152036970R582). Aren't they supposed to do the same thing? Am I missing something here?

Reply:
that condition is for new_password_encryption and confirm_password_encryption fields, the line that you're pointing out is for the password_encryption field which stores the confirmed encryption password.